src/winforms/toga_winforms/window.py

```diff
@@ -111,6 +112,11 @@ def show(self):
         if self.interface is not self.interface.app._main_window:
             self.native.Show()
 
+    def on_formclose(self, event, handler):
+        if self.interface.app.on_exit:
+            self.interface.app.on_exit(self.interface.app)
+        print('On form close')
```

Comment:
This version of the winforms signal handler (invoking the app on_exit) should probably be on the MainWindow, rather than on *every* window. As written, any multi-window app would run it's exit logic if any window was closed.

(Also - stray print statement!)

Reply:
Definitely. I added checking for main window, and, to do that, put that line into `show()`, because on `create` the app is still None.
Thank you for your patience. I ran a search for print statements, but I guess I missed one. I worked on this for a couple of hours, trying to get it to work and changing places of functions, (window and app) so, I guess I missed it somewhere along the way.

Should I change the windows-specific handler names in other widgets?